src/Swarm/Game/World/Parse.hs

```diff
@@ -0,0 +1,264 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE ViewPatterns #-}
+-- FromJSON WExp
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Parser for the Swarm world description DSL.
+module Swarm.Game.World.Parse where
+
+import Control.Monad (MonadPlus, void)
+import Control.Monad.Combinators.Expr (Operator (..), makeExprParser)
+import Data.List.NonEmpty (NonEmpty)
+import Data.List.NonEmpty qualified as NE
+import Data.Text (Text)
+import Data.Text qualified as T
+import Data.Void
+import Data.Yaml (FromJSON (parseJSON), withText)
+import Swarm.Game.World.Syntax
+import Swarm.Util (failT, showT, squote)
+import Text.Megaparsec hiding (runParser)
+import Text.Megaparsec.Char
+import Text.Megaparsec.Char.Lexer qualified as L
+import Witch (into)
+
+type Parser = Parsec Void Text
+type ParserError = ParseErrorBundle Text Void
+
+------------------------------------------------------------
+-- Utility
+
+sepByNE :: (MonadPlus m) => m a -> m sep -> m (NonEmpty a)
+sepByNE p sep = NE.fromList <$> p `sepBy1` sep
+
+------------------------------------------------------------
+-- Lexing
+
+reservedWords :: [Text]
+reservedWords =
+  [ "not"
+  , "true"
+  , "false"
+  , "seed"
+  , "x"
+  , "y"
+  , "hash"
+  , "let"
+  , "in"
+  , "overlay"
+  , "hcat"
+  , "vcat"
+  , "if"
+  , "then"
+  , "else"
+  , "perlin"
+  , "mask"
+  , "empty"
+  , "abs"
+  ]
+
+-- | Skip spaces and comments.
+sc :: Parser ()
+sc =
+  L.space
+    space1
+    (L.skipLineComment "//")
+    (L.skipBlockComment "/*" "*/")
```

Comment:
I don't see any comments in the classic world or other world description while scrolling through.

Maybe we could have a simpler training world with comments that note the basic features? 🙂

Reply:
Good idea, I'll add both.

Reply:
Added some comments to the classic world and created a `README` in `data/worlds` with a full description of the DSL.